tapgarden/custodian_test.go

```diff
@@ -402,21 +531,92 @@ func TestTransactionHandling(t *testing.T) {
 
 	// Only one event should be registered though, as we've only created one
 	// transaction.
-	h.eventually(func() bool {
-		events, err := h.tapdbBook.QueryAddrEvents(
-			ctx, address.EventQueryParams{},
-		)
-		require.NoError(t, err)
+	events := h.assertEventsPresent(1, address.StatusCompleted)
+	require.EqualValues(t, outputIdx, events[0].Outpoint.Index)
 
-		if len(events) != 1 {
-			t.Logf("Got %d events", len(events))
-			return false
-		}
+	dbProof, err := h.assetDB.FetchProof(ctx, mockProof.Locator)
+	require.NoError(t, err)
+	require.EqualValues(t, mockProof.Blob, dbProof)
+}
+
+// TestTransactionConfirmedOnly tests that the custodian only starts the proof
+// courier once a transaction has been confirmed. We also test that it correctly
+// re-tries fetching proofs using a proof courier after it has been restarted.
+func TestTransactionConfirmedOnly(t *testing.T) {
```

Comment:
💯 

This serves as a repro for #597 right?

Reply:
Ah yes, correct.